src/utils.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
```

Comment:
Can you use promisify instead of these functions? https://github.com/digitaldesignlabs/es6-promisify#readme

Reply:
I added these as an interim solution. We're going to be doing a lot of work soon for they async iterators initiative https://github.com/libp2p/js-libp2p/issues/266 which will remove the need for these or promisfy. If that ends up taking longer than we'd like, or we have more immediate needs in other parts of the code base, then I think switching over to promisify is warranted.

Reply:
👍